docs/reference/migration/transient-settings-migration-guide.asciidoc

```diff
@@ -0,0 +1,107 @@
+[[transient-settings-migration-guide]]
+=== Transient settings migration guide
+
+////
+[source,console]
+----
+PUT _cluster/settings
+{
+  "transient": {
+    "cluster.indices.close.enable": false,
+    "indices.recovery.max_bytes_per_sec": "50mb"
+  }
+}
+----
+// TEST[warning:[transient settings removal] Updating cluster settings through transientSettings is deprecated. Use persistent settings instead.]
+////
+
+Transient cluster settings are deprecated. Transient settings don't survive full
+cluster restarts. A cluster restart can occur unexpectedly if enough
+master-eligible nodes fail, resulting in an unwanted cluster configuration.
```

Comment:
> A cluster restart can occur unexpectedly if enough
master-eligible nodes fail, resulting in an unwanted cluster configuration.

I am not sure I follow that sentence. I think it means under some circumstances transient settings may not survive cluster instability  ? (I am not too familiar with this area of the code to say with certainty)

Reply:
Thanks for the feedback, Jake. That's the essence of what I'm saying, but I'll revise the following so it's clearer:

> Transient cluster settings are deprecated. Previously, you could use transient
settings to make temporary configuration changes to a cluster. However,
transient settings don't survive full cluster restarts. A cluster restart can
occur forcibly if enough master-eligible nodes fail. This can clear transient
settings unexpectedly, leading to a potentially undesired cluster configuration.

Credit to @grcevski. I'm paraphrasing his work in the deprecation docs here.

Reply:
> Previously, you could use transient
settings to make temporary configuration changes to a cluster. However,
transient settings don't survive full cluster restarts. A cluster restart can
occur forcibly if enough master-eligible nodes fail. This can clear transient
settings unexpectedly, leading to a potentially undesired cluster configuration.

I think " A cluster restart can occur forcibly if" is overloading the term cluster restart. Can we call this cluster instability ?

For example:

However,
transient settings don't survive full cluster restarts or cluster instability.
If a cluster is restarted or is unstable then transient
settings can be cleared unexpectedly, leading to a potentially undesired cluster configuration.

Reply:
👍  Thanks for the feedback. I've reworded to:

> Transient cluster settings are deprecated. Previously, you could use transient
settings to make temporary configuration changes to a cluster. However, a
cluster restart or cluster instability can unexpectedly clear these settings,
leading to a potentially undesired cluster configuration.